hcl2template/types.packer_config.go

```diff
@@ -391,6 +488,8 @@ func (cfg *PackerConfig) GetBuilds(opts packer.GetBuildsOptions) ([]packer.Build
 			pcb.Provisioners = provisioners
 			pcb.PostProcessors = pps
 			pcb.Prepared = true
+			pcb.HCL2ProvisionerPrepare = cfg.HCL2ProvisionerPrepare
```

Comment:
This could be its own type, like:
 
```suggestion
			pcb.HCL2ProvisionerPrepare = provisionersPreparer{cfg, provisioners}.Func
```

This way we wont have to do that scary look up inside HCL2ProvisionerPrepare.

Reply:
`provisioners` you mean the `build.ProvisionerBlocks` right? It can't be the `provisioners` because this is already decoded with the unknown value.

Reply:
Well, I think both, I'm on the team "let's re-prepare those provisioners" and I also don't think it's bad that they were previously prepared since those value will simply be replaced.

Reply:
I did this idea of the struct and the code looks muuuuch better!

Reply:
Nice !!